process/sync/baseSync.go

```diff
@@ -154,67 +157,49 @@ func (boot *baseBootstrap) processReceivedHeader(headerHandler data.HeaderHandle
 		log.Debug("forkDetector.AddHeader", "error", err.Error())
 	}
 
-	boot.mutRcvHdrHash.Lock()
-	hash := boot.requestedHeaderHash()
-	if hash == nil {
-		boot.mutRcvHdrHash.Unlock()
-		return
-	}
+	go boot.requestMiniBlocks(headerHandler.GetShardID(), headerHandler.GetNonce())
+
+	boot.computeReceivedHeaderByNonce(headerHandler, headerHash)
+	boot.computeReceivedHeaderByHash(headerHandler, headerHash)
+}
 
-	if bytes.Equal(hash, headerHash) {
+func (boot *baseBootstrap) computeReceivedHeaderByNonce(headerHandler data.HeaderHandler, hdrHash []byte) {
+	boot.mutRcvHdrNonce.Lock()
+	n := boot.requestedHeaderNonce()
+	if n != nil && *n == headerHandler.GetNonce() {
 		log.Debug("received requested header from network",
 			"shard", headerHandler.GetShardID(),
 			"round", headerHandler.GetRound(),
 			"nonce", headerHandler.GetNonce(),
-			"hash", hash,
+			"hash", hdrHash,
 		)
-		boot.setRequestedHeaderHash(nil)
-		boot.mutRcvHdrHash.Unlock()
-		boot.chRcvHdrHash <- true
-	} else {
-		boot.mutRcvHdrHash.Unlock()
-	}
-}
+		boot.setRequestedHeaderNonce(nil)
+		boot.mutRcvHdrNonce.Unlock()
+		boot.chRcvHdrNonce <- true
 
-// receivedHeaderNonce method is a call back function which is called when a new header is added
-// in the block headers pool
-func (boot *baseBootstrap) receivedHeaderNonce(nonce uint64, shardId uint32, hash []byte) {
-	if boot.shardCoordinator.SelfId() != shardId {
 		return
 	}
 
-	log.Trace("received header from network",
-		"shard", shardId,
-		"nonce", nonce,
-		"hash", hash,
-	)
-
-	err := boot.addReceivedHeaderToForkDetector(hash)
-	if err != nil {
-		log.Debug("addReceivedHeaderToForkDetector", "error", err.Error())
-	}
-
-	go boot.requestMiniBlocks(shardId, nonce)
-
-	boot.mutRcvHdrNonce.Lock()
-	n := boot.requestedHeaderNonce()
-	if n == nil {
-		boot.mutRcvHdrNonce.Unlock()
-		return
-	}
+	boot.mutRcvHdrNonce.Unlock()
+}
 
-	if *n == nonce {
+func (boot *baseBootstrap) computeReceivedHeaderByHash(headerHandler data.HeaderHandler, hdrHash []byte) {
```

Comment:
wrong name for this func. Duplicate code with the function computeReceivedHeaderByNonce

Reply:
method was renamed.